packages/pushnotification/android/src/main/java/com/amazonaws/amplify/pushnotification/RNPushNotificationMessagingService.java

```diff
@@ -167,4 +168,54 @@ private Bundle convertDataToBundle(RemoteMessage message) {
         return bundle;
     }
 
+	/**
+	 * Called if InstanceID token is updated. This may occur if the security of
+	 * the previous token had been compromised. Note that this is called when the InstanceID token
+	 * is initially generated so this is where you would retrieve the token.
+	 */
+	@Override
+	public void onNewToken(String refreshedToken) {
+		// Get updated InstanceID token.
+		Log.i(LOG_TAG, "Refreshed token: " + refreshedToken);
+
+		final Bundle bundle = createBundleWithToken(refreshedToken);
+		// We need to run this on the main thread, as the React code assumes that is true.
```

Comment:
This is specific to React Native correct? 

```suggestion
		// We need to run this on the main thread, as the React Native code assumes that is true.
```

Reply:
That's right. This function has been moved from the other class that was used by the previous firebase version called `RNPushNotificationDeviceIDService`. 
And yeah you are right. It is React Native specific. Thanks!